### PR TITLE
Update submodule

### PR DESCRIPTION
**Pull request by yunhaoling**
update to azure-uamqp-c commit:
https://github.com/Azure/azure-uamqp-c/tree/259db533a66a8fa6e9ac61c39a9dae880224145f

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ Release History
 +++++++++++++++++++
 
 - `SASTokenAuth`, `JWTTokenAuth`, `SASTokenAsync`, and `JWTTokenAsync` now takes keyword argument `refresh_window` to override default token refresh timing in constructors.
+- Updated Azure uAMQP C and Azure C Shared Utility dependencies.
```
Comment (lmazuel) [inline HISTORY.rst:10]: Are those versionned that we could it there? for clarity? Or a date at least when you copy them?

Comment (lmazuel) [reply HISTORY.rst:10]: Also, any changelog they wrote so we can know what is fixed?

**Review by lmazuel — APPROVED** [review]
Sounds fair

File: HISTORY.rst (hunk 1 above)
Comment (yunhaoling) [reply HISTORY.rst]: good suggestion! added date & commit information

**Pull request merged**